aten/src/ATen/cuda/CachingHostAllocator.cpp

```diff
@@ -156,6 +156,38 @@ struct CUDACachingHostAllocatorImpl
     return true;
   }
 
+  bool is_pinned(void* ptr) override {
+    // First check if driver is broken/missing, in which case PyTorch CPU
+    // functionalities should still work, we should report `false` here.
+    if (!at::cuda::is_available()) {
```

Comment:
can this call itself cause driver initialization?

Reply:
Shouldn't we check for something like `is_initialized`? Because if driver is not initialized, ptr is obviously not pinned?

Reply:
Got it. I will try to improve it.

Reply:
@malfet The logic is already handled at https://github.com/pytorch/pytorch/blob/ad11d6378c74a86ea084ca6bff8d79f7ec3e24ac/aten/src/ATen/Context.h#L104
However, I don't think this is a good place since it duplicates [is_initialized](https://github.com/pytorch/pytorch/blob/ad11d6378c74a86ea084ca6bff8d79f7ec3e24ac/torch/csrc/utils/device_lazy_init.cpp#L16). Do you think we should merge them into one?

Reply:
@malfet May I know if I have addressed your comments